File: myexamples/imProcessingBench/bench_matching.py
# Template Matching

import sys
import numpy as np
import cv2
import time

NTIMES=1000

if __name__ == '__main__':
	if len(sys.argv)!=3:
		print "usage imgfilename tempfilename"
		sys.exit(1)
	img = cv2.imread(sys.argv[1],0)
	tmp = cv2.imread(sys.argv[2],0)

	width,height = tmp.shape[::-1] 
	processingTime=0
	for i in range(0, NTIMES):
		sttime=time.time()
		res = cv2.matchTemplate(img,tmp,cv2.TM_CCOEFF_NORMED)
		edtime=time.time()
		processingTime+=(edtime-sttime)
		min_val, max_val, min_loc, max_loc = cv2.minMaxLoc(res)
		top_left = max_loc
		bottom_right = [top_left[0]+width, top_left[1]+height]
	print "matched at ( "+str(top_left[0])+" , "+str(top_left[1])+" ) to ( "+str(bottom_right[0])+" , "+str(bottom_right[1])+" )."
	print "Time "+str(processingTime)+"ms"

File: myexamples/imProcessingBench/bench_matching.cpp
//Template Matching

#include <iostream>
#include <vector>
#include <chrono>

#include "opencv2/imgproc/imgproc.hpp"
#include "opencv2/highgui/highgui.hpp"

#define NTIMES 1000

using namespace std;
using namespace cv;

int main(int argc, char* argv[]){
	Mat img,tmp,result;
	Point top_left;
	double proccessingTime=0;
	if(argc!=3){
		cerr<<"usage imgfilename tmpfilename"<<endl;
		exit(1);
	}

	img=imread(argv[1]);
	tmp=imread(argv[2]);
	int width=tmp.cols;
	int height=tmp.rows;

	for(int i=0;i<NTIMES;i++){
		auto start=std::chrono::system_clock::now();
		matchTemplate( img, tmp, result, TM_CCOEFF_NORMED);
		auto end=std::chrono::system_clock::now();
		proccessingTime+=std::chrono::duration_cast<std::chrono::milliseconds>
		/// Localizing the best match with minMaxLoc
	  	double minVal, maxVal; 
	  	Point minLoc, maxLoc;
	  	minMaxLoc( result, &minVal, &maxVal, &minLoc, &maxLoc, Mat() );
	  	top_left= maxLoc;
	}
	cerr<<"matched at ( "<<top_left.x<<", "<<top_left.y<<" ) to ( "<<(top_left.x+width)<<" , "<<(top_left.y+height)<<" )."<<endl;
	cout<<"Time "<<proccessingTime<<"ms"<<std::endl;
	return 0;
}